### DISCUSSION TITLE
SenseCAP S2100 LoRaWAN Data Logger | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#237] [Q&A]
# SenseCAP S2100 LoRaWAN Data Logger | Seeed Studio Wiki

SenseCAP S2100 LoRaWAN Data Logger

https://wiki.seeedstudio.com/Sensor/SenseCAP/SenseCAP_LoRaWAN_Sensor/SenseCAP_S210X_Series/tutorial/Connect-S210X-Sensor-to-Microsoft-Azure-IoT-Central

**Comment by merrittholmes7** [comment]
Does anyone know where I can get support for this as there are some fundamental flaws in the product. Namely it cannot support anything except a number output from a register and you cannot specify the length of the register value. It also doesn't allow you to see what message was sent to the device to read the register for debugging.

**Reply by Zeke5579** [reply]
We're here to help! You can reach us at sensecap@seeed.cc. Regarding the output format and data, we've got you covered! The reason the output cannot be changed is because it is necessary to work with the decoder on the server to decode the correct information from the S2100 output data!

**Comment by M10CUBE** [comment]
I am curious. Why the support team has no answer for the above Q placed by merrittholmes7? I am going to get one and if the above is the case it is a problem. 
Thank you

**Reply by Zeke5579** [reply]
We'd love to hear from you! Please don't hesitate to reach out to us at sensecap@seeed.cc.

**Comment by merrittholmes7** [comment]
And why can't the decode decode a normal string or different date formats. The product is not fit for purpose with these massive restrictions. The whole point is that you can log data from different devices and so it needs to be able to handle different types of strings, dates, numbers. On 10 Dec 2024 01:28, Zeke5579 ***@***.***> wrote:
We're here to help! You can reach us at ***@***.*** Regarding the output format and data, we've got you covered! The reason the output cannot be changed is because it is necessary to work with the decoder on the server to decode the correct information from the S2100 output data!

**Reply by Zeke5579** [reply]
LoRaWAN cannot directly transmit plain strings or varied date formats due to:
1. Bandwidth Limitations: Small payload sizes make lengthy data impractical.
2. Binary Payloads: LoRaWAN transmits binary data; encoding/decoding is required at the application layer.

Our pre-set data formats guarantee the reliability of information transmission. If you have a large number of data format customization needs, contact us at sensecap@seeed.cc. We will customize the data format you need within the framework of the Lorawan agreement.

**Comment by merrittholmes7** [comment]
Yes of course I understand how LoRaWAN works, but you can still encode an 8bit string or say a mac address into a LoRaWAN payload but its impossible to decode this with your device. Similarly, its impossible to decode timestamps. quite often the device id will be the payload. This is also impossible to decode as there is no string type available.
________________________________

**Reply by Zeke5579** [reply]
For your query, our devices just send the captured information to the gateway, which transfers the information to the platform and then decodes it, if you have a batch of devices that need to be loaded with specific information, we offer customization services to meet your needs.

**Comment by merrittholmes7** [comment]
The problem is that sometimes the captured Information is a string or a date and you have no way of decoding as an option. This is a product limitation which is very very basic. On 19 Dec 2024 03:35, Zeke5579 ***@***.***> wrote:
For your query, our devices just send the captured information to the gateway, which transfers the information to the platform and then decodes it, if you have a batch of devices that need to be loaded with specific information, we offer customization services to meet your needs.

**Reply by Zeke5579** [reply]
Thank you so much for your ideas! We'll absolutely take them into consideration during ours product development. Thank you so much for supporting our development and making suggestions!

**Comment by bprfh** [comment]
Does the datalogger buffer measurements when offline, or does it discard measurements if there is no network?

**Reply by nfs0619** [reply]
Hello, the S2100 does not have a data buffering function. If the device is offline, measurements will not be stored and will be lost.

**Comment by Devilfish75** [comment]
hello i have this logger for my weatherstation s1000 V2 . i need help where are in the manuals the last 3 Measurement Setting. the s1000 has 10 not 7. please help!

**Reply by Devilfish75** [reply]
Set each measurement in turn. Measurement1 (Air temperature)

Register Address	The register address of the measured value in the sensor, which is an integer. The Air temperature register address is 0.Enter 0 here.
Function Code	Modbus function code, select 03 here.
Data Type	
The data type determines the number of registers read from the sensor and how the data should parse the value.

Select Signed 32bit integer, 0xABCD here.

Precision	
Precision of the value. You can choose the decimal place of the measurement value. If 1 is selected, one decimal place is reserved.

Select 2, #.## here.

Y= Ax + B	
“Y”: It is the value of Data Logger will upload.

“x”: It is the original current value.

Factory A: Custom values that can be scaled up or down by multiples of the “x”.

Factory B: A custom value that increments or diminishes the value of the “x”.

By setting the values of A and B, you can calculate the desired value. If only raw values are uploaded, set A=1 and B=0.

Factory A is 0.001 and Factory B is 0.

Write Strategy	
This function is enabled only for some special sensors and is generally disabled by default

After reading the value of the register, special instructions can be issued to the sensor, such as the instruction to empty the register after reading register 0.

None: Off by default.

After Read: Send the RS485 command to sensor after reading the register.

On New Data: Send the RS485 command to sensor every 24 hours.

We don’t need to write strategy so we select None here.

Measurement2 (Air humidity)

Register Address	Enter 2 here.
Function Code	Select 03 here.
Data Type	Select Signed 32bit integer, 0xABCD here.
Precision	Select 2, #.## here.
Y= Ax + B	Factory A is 0.001 and Factory B is 0.
Write Strategy	We don’t need to write strategy so we select None here.
Measurement3 (Barometric pressure)

Register Address	Enter 4 here.
Function Code	Select 03 here.
Data Type	Select Signed 32bit integer, 0xABCD here.
Precision	Select 0, # here.
Y= Ax + B	Factory A is 0.001 and Factory B is 0.
Write Strategy	We don’t need to write strategy so we select None here.
Measurement4 (Light intensity)

Register Address	Enter 6 here.
Function Code	Select 03 here.
Data Type	Select Signed 32bit integer, 0xABCD here.
Precision	Select 0, # here.
Y= Ax + B	Factory A is 0.001 and Factory B is 0.
Write Strategy	We don’t need to write strategy so we select None here.
Measurement5 (Average wind direction)

Register Address	Enter 12 here.
Function Code	Select 03 here.
Data Type	Select Signed 32bit integer, 0xABCD here.
Precision	Select 1, #.# here.
Y= Ax + B	Factory A is 0.001 and Factory B is 0.
Write Strategy	We don’t need to write strategy so we select None here.
Measurement6 (Average wind speed)

Register Address	Enter 18 here.
Function Code	Select 03 here.
Data Type	Select Signed 32bit integer, 0xABCD here.
Precision	Select 1, #.# here.
Y= Ax + B	Factory A is 0.001 and Factory B is 0.
Write Strategy	We don’t need to write strategy so we select None here.
Measurement7 (Rain intensity)

Register Address	Enter 24 here.
Function Code	Select 03 here.
Data Type	Select Signed 32bit integer, 0xABCD here.
Precision	Select 1, #.# here.
Y= Ax + B	Factory A is 0.001 and Factory B is 0.
Write Strategy	We don’t need to write strategy so we select None here.
After configuring these measurements, click "Back to Home". The node and the app Bluetooth will automatically disconnect. The Data Logger will attempt to connect to the network. The LED indicator will flash red slowly while trying to connect and will flash green quickly once the network connection is successful.

**Reply by Devilfish75** [reply]
https://wiki.seeedstudio.com/Sensor/SenseCAP/SenseCAP_Data_Logger/tutorial/How_to_Configure_the_12V_RS485_Sensor_for_S2100_Data_Logger/#measurement-setting

**Reply by 1cecO1** [reply]
Hi, please refer to page 6 of the user manual to view all the measurement values ​​of S1000, and refer to the table in 4.1.2 Register Address Definition on page 28 to find the corresponding register address and then configure it in S2100. If you have further questions, please contact us at techsupport@seeed.io
<img width="2493" height="1357" alt="20260420-163008" src="https://github.com/user-attachments/assets/ca508a63-272c-4742-9e55-493a15af4e8f" />

**Reply by 1cecO1** [reply]
**User Manual:** https://files.seeedstudio.com/products/SenseCAP/SenseCAP_ONE/SenseCAP_ONE_V3_Compact_Weather_Station_User_Guide_V1.6.pdf
